src/NRediSearch/Client.cs

```diff
@@ -50,19 +50,104 @@ public enum IndexOptions
             KeepTermFrequencies = 16
         }
 
+        public sealed class IndexDefinition
+        {
+            public enum IndexType
+            {
+                /// <summary>
+                /// Used to indicates that the index should follow the keys of type Hash changes
+                /// </summary>
+                HASH
```

Comment:
minor naming nit; .NET standards would make this `Hash` (if you need `HASH` for some text purposes, that can be done via translation in any of a number of ways); note also that .NET enums are not checked - and any value (`42`, for example) can be passed in; you can tweak both of these in a simple way - will add example below (in the "add args" step)

Reply:
Thanks, I'll rename, there is no casing thing here, `Hash` will do.